src/OrchardCore.Modules/OrchardCore.AuditTrail/Services/AuditTrailManager.cs

```diff
@@ -116,7 +116,7 @@ public Task<AuditTrailEvent> GetEventAsync(string eventId) =>
 
     public async Task<int> TrimEventsAsync(TimeSpan retentionPeriod)
     {
-        var dateThreshold = _clock.UtcNow.AddDays(1) - retentionPeriod;
```

Comment:
Why you changed from what it was?

Reply:
Because there is a `rententionPeriod` that tells how long ago to take. Why add 1 day to it? If I say I want to remove the last 10h, why would it take 1 day and 10h?

Reply:
It's even worse than that, since this adds a day, i.e. it wouldn't take anything (it would take 14 hours from the future).